odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2826,6 +2826,23 @@ def process(node, info, prefix):
         process(etree.fromstring(view_info['arch']), view_info, '')
         return result
 
+    def _ensure_xml_id(self):
+        # Prevent triggering unique constraint when updating a module where master data was added
+        # both in the code and by a user.
+        # i.e. Country state was added both in the code and in master data => shouldn't trigger
+        self.env['ir.model.data']._update_xmlids([
+            {
+                'xml_id': record._xml_id_builder(),
+                'record': record,
+                'noupdate': True,
+            }
+            for record in (
+                self.browse(meta['id'])
+                for meta in self.search([]).get_metadata()
+                if not meta.get('xmlid')
```

Comment:
If the records were created through an import (or were exported with import compatible export), the records will already have an XMLID that won't match the one returned by the builder.
So to be 100% reliable, we need to check if each record is missing the XMLID matching the one returned by the builder.
```suggestion
                self.browse(
                    [
                        state.id for state in self.search([])
                        if not [
                            meta for meta in state.get_metadata()
                            if meta.get("xmlid") == state._xml_id_builder()
                        ]
                    ]
                )
```